backend/src/controllers/traineeController.ts

```diff
@@ -0,0 +1,162 @@
+import { Response } from "express";
+import { editUserSchema } from "../validations/userValidation";
+import User from "../models/User";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+export const getTrainees = async (req: any, res: Response) => {
+  const searchString = req.query.searchString || "";
+  const traineesPerPage = Number(req.query.coachesPerPage) || 10;
+  const sortBy = req.query.sortBy || "entry";
+  try {
+    const trainees = await User.aggregate([
+      {
+        $match: {
+          $or: [{ name: { $regex: new RegExp(searchString, "i") } }],
+          role: "TRAINEE",
+        },
+      },
+      {
+        $lookup: {
+          from: "users",
+          localField: "coach",
+          foreignField: "_id",
+          as: "coach",
+        },
+      },
+      {
+        $addFields: {
+          coach: {
+            $cond: {
+              if: { $eq: [{ $size: "$coach" }, 0] },
+              then: [{}],
+              else: "$coach",
+            },
+          },
+        },
+      },
+      {
+        $project: {
+          _id: 1,
+          name: 1,
+          email: 1,
+          role: 1,
+          coach: {
+            $cond: {
+              if: { $eq: [{ $size: "$coach" }, 0] },
+              then: {},
+              else: {
+                $arrayElemAt: ["$coach", 0],
+              },
+            },
+          },
+        },
+      },
+      {
+        $sort: { [sortBy]: 1 },
+      },
+      {
+        $limit: traineesPerPage,
+      },
+    ]);
+
+    return res.status(200).json(trainees);
+  } catch (error) {
+    res.status(400).send("failed to get trainees ");
+  }
+};
+export const getTraineesForCoach = async (req: any, res: Response) => {
```

Comment:
Add an empty line between methods pls ... @NNesta maybe you can setup a prettier and that will automatically format the files for you guys.